Geometry.cs

```diff
@@ -124,36 +124,70 @@ class Polygon {
          p0 = p1;
       }
    }
+
 }
 
 /// <summary>A drawing is a collection of polygons</summary>
 class Drawing {
    public void Add (Polygon poly) {
       mPolys.Add (poly);
-      mBound = new (); 
+      if (mConvexEnvelope is not null) mConvexEnvelope = GetEnvelope (mConvexEnvelope.Concat (poly.Pts));
+      mBound = new ();
    }
 
    public IReadOnlyList<Polygon> Polys => mPolys;
    List<Polygon> mPolys = new ();
 
    public static Drawing operator * (Drawing d, Matrix2 m) {
-      Drawing d2 = new Drawing ();
+      Drawing d2 = new ();
       foreach (var p in d.Polys) d2.Add (p * m);
       return d2;
    }
-
-   public Bound2 Bound {
-      get {
-         if (mBound.IsEmpty) mBound = new (Polys.Select (a => a.Bound));
-         return mBound;
-      }
-   }
    Bound2 mBound;
 
-   public Bound2 GetBound (Matrix2 xfm) 
-      => new Bound2 (Polys.SelectMany (a => a.Pts.Select (p => p * xfm)));
+   public Bound2 GetBound (Matrix2 xfm)
+      => new (ConvexEnvelope.Select (a => a * xfm));
 
    /// <summary>Enumerate all the lines in this drawing</summary>
-   public IEnumerable<(Point2 A, Point2 B)> EnumLines (Matrix2 xfm) 
+   public IEnumerable<(Point2 A, Point2 B)> EnumLines (Matrix2 xfm)
       => mPolys.SelectMany (a => a.EnumLines (xfm));
-}
+
+   /// <summary>Enumerate all the lines enclosing the convex polygon in this drawing</summary>
+   public IEnumerable<(Point2 A, Point2 B)> EnumEnvelopeLines (Matrix2 xfm) {
+      var Hull = ConvexEnvelope;
+      var p0 = Hull[^1] * xfm;
+      foreach (var p in Hull) {
+         var p1 = p * xfm;
+         yield return (p0, p1);
+         p0 = p1;
+      }
+   }
+
+   public IReadOnlyList<Point2> ConvexEnvelope {
+      get {
+         mConvexEnvelope ??= GetEnvelope (Polys.SelectMany (a => a.Pts));
+         return mConvexEnvelope;
+      }
+   }
+   IReadOnlyList<Point2> mConvexEnvelope;
+
+   /// <summary>Computes the convex envelope lines of given set of polygon points (using Graham scan algorithm)</summary>
+   IReadOnlyList<Point2> GetEnvelope (IEnumerable<Point2> pts) {
+      var bottomPt = pts.MinBy (p => p.Y);
+      var sortedPts = pts.OrderBy (a => a.AngleTo (bottomPt)).ToList ();
+      sortedPts.Remove (bottomPt);
+      sortedPts.Prepend (bottomPt);
+      var HullPts = new List<Point2> ();
+      for (int i = 0; i < sortedPts.Count; i++) {
+         var nextPt = sortedPts[i];
+         while (HullPts.Count > 2) {
+            if (IsObtuse (HullPts[^2], HullPts[^1], nextPt)) break;

```

Comment:
For a single line method, unless you are using it in another location, local function is not needed. As this will pass three point to function and return which is not needed.